integrations/salesforce/salesforceauth/salesforceauth.go

```diff
@@ -0,0 +1,66 @@
+package salesforceauth
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
+
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+// AccessTokenExpiration sends a request to the given instance's OAuth2 introspection endpoint to retrieve
+// the expiration timestamp of the token and updates the provided token's expiry field.
+// Returns an error if the request fails, the response is invalid, or the expiration timestamp is missing.
+func AccessTokenExpiration(ctx context.Context, instanceURL string, t *oauth2.Token, vsid sdktypes.VarScopeID, vars sdkservices.Vars) error {
+	vs, err := vars.Get(ctx, vsid)
+	if err != nil {
+		return err
+	}
+
+	formData := url.Values{
+		"token":           {t.AccessToken},
+		"token_type_hint": {"access_token"},
+		"client_id":       {vs.GetValueByString("private_client_id")},
+		"client_secret":   {vs.GetValueByString("private_client_secret")},
```

Comment:
Check that `private_client_id` and `private_client_secret` are not empty? (unless they are optional in the request).